common/s2n-codec/src/decoder/checked_range.rs

```diff
@@ -76,3 +77,25 @@ impl fmt::Debug for CheckedRange {
         write!(f, "{}..{}", self.start, self.end)
     }
 }
+/// The following harnesses were automatically generated by an
+/// experimental tool developed by the Kani team.
+#[cfg(test)]
+mod bolero_harnesses {
+    use super::*;
+    #[test]
+    #[cfg_attr(kani, kani::proof)]
+    fn bolero_test_25_len() {
+        bolero::check!()
+            .with_type()
+            .cloned()
```

Comment:
Do these really need to be `.cloned()`?

Reply:
I've removed some of the usage of `cloned()` in the tests that allow it. Currently, the tool always uses `cloned()` when generating bolero harnesses. 

Is there a way to tell at compile-time if we need to use `cloned()` or not?

Reply:
> Is there a way to tell at compile-time if we need to use cloned() or not?

You'd have to be able to query that in your tool, which would probably be complicated :). I don't think it's the end of the world to have `.cloned()` on everything. It can just mean that the fuzzer runs a bit slower. Kani shouldn't matter, since the `for_each` does actually run in a loop.